pkg/cloudinit/metadata.go

```diff
@@ -32,12 +35,14 @@ type Metadata struct {
 }
 
 // NewMetadata returns a new Metadata object.
-func NewMetadata(instanceID, hostname string, injectProviderID bool) *Metadata {
+func NewMetadata(instanceID, hostname string, kubernetesVersion string, injectProviderID bool, injectKubernetesVersion bool) *Metadata {
```

Comment:
Could we do without both `kubernetesVersion` and `injectKubernetesVersion`? You could set `BaseCloudInitData.KubernetesVersionInjection` to true or false depending on whether `kubernetesVersion` is unset (say, empty string, or nil pointer).

If not, I would prefer the two args be kept together, like so
`func NewMetadata(instanceID, hostname string, injectProviderID bool, injectKubernetesVersion bool, kubernetesVersion string) *Metadata {`

Reply:
Thanks for your feedback!

I have removed the `injectKubernetesVersion` field. The `kubernetes-version` will now be injected into the metadata if `spec.version` is set on the machine.

I had to add a new argument to `NewMetadata` because I don't have access to the `MachineScope` there and don't know of any other way to get the Kubernetes version here.

What do you think of the new, much simpler implementation? I removed the draft flag because I think it is ready now. I have successfully tested the implementation on our infrastructure.